File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/exception/LenderExceptionFactory.java
package com.nivasafinance.features.lender.lender.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

import java.util.UUID;

public class LenderExceptionFactory {

    public static LenderOperationException createFailed(MessageSource messageSource) {
        return new LenderOperationException("error.lender.operation.create", messageSource);
    }

    public static LenderOperationException retrieveFailed(MessageSource messageSource) {
        return new LenderOperationException("error.lender.operation.retrieve", messageSource);
    }

    public static LenderOperationException duplicateKey(String key, MessageSource messageSource) {
        return new LenderOperationException(
                ExceptionUtils.createLocalizedMessage("error.lender.duplicate.key", new Object[]{key}, messageSource));
    }

    public static LenderNotFoundException lenderNotFound(UUID lenderId, MessageSource messageSource) {
        return new LenderNotFoundException(lenderId, messageSource);
    }

    public static LenderKeyNotFoundException lenderNotFound(String key, MessageSource messageSource) {
        return new LenderKeyNotFoundException(key, messageSource);
    }
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/exception/LenderExceptionFactory.kt
package com.nivasafinance.features.lender.lender.exception

import org.springframework.context.MessageSource
import java.util.UUID

object LenderExceptionFactory {

    fun createFailed(messageSource: MessageSource): LenderOperationException {
        return LenderOperationException("error.lender.operation.create", messageSource)
    }

    fun lenderNotFound(lenderId: UUID, messageSource: MessageSource): LenderNotFoundException {
        return LenderNotFoundException(lenderId, messageSource)
    }

    fun lenderNotFound(key: String, messageSource: MessageSource): LenderKeyNotFoundException {
        return LenderKeyNotFoundException(key, messageSource)
    }
}
